api/src/dtos/applications/public-apps-view-params.dto.ts

```diff
@@ -0,0 +1,24 @@
+import { Expose, Type } from 'class-transformer';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsEnum, IsString } from 'class-validator';
+import { ValidationsGroupsEnum } from '../../enums/shared/validation-groups-enum';
+import { ApplicationsFilterEnum } from '../../enums/applications/filter-enum';
+
+export class PublicAppsViewQueryParams {
+  @Expose()
+  @ApiProperty({
+    type: String,
+    example: 'userId',
+  })
+  @IsString({ groups: [ValidationsGroupsEnum.default] })
+  userId: string;
+
+  @Expose()
+  @ApiPropertyOptional({
+    enum: ApplicationsFilterEnum,
+    enumName: 'ApplicationsFilterEnum',
+    example: 'all',
+  })
+  @IsEnum(ApplicationsFilterEnum, { groups: [ValidationsGroupsEnum.default] })
+  filterType?: ApplicationsFilterEnum;
```

Comment:
This isn't a problem because of this exact line, just commenting here as it is relevant to it. With the frontend passing the filtering type we still have the small problem of the page reloading whenever you click a different status on the left nav bar. This causes a flash of all of the numbers and it's not as smooth of an experience.

Would it be possible to always return all applications and do the filtering on the frontend? Especially now that the returned type from the backend is a lot smaller we don't have too much of a concern of there being too much data. Or in other words we only call the backend on initial load of the page and then when we go to the different tabs it filters for the correct applications. I do remember you talking about there being limitations on the UI component and accessibility so would be curious to get your thoughts on this

Reply:
Totally agree on it not being super smooth. I think if the long term plan was to stick with the Tabs component, then I would go for the FE filtering experience but since it is a temporary solution until the menu component is ready (which relies on links) and Jesse is advising that separate pages are the a11y best practice, I'm inclined to keep this approach as is. I did also have some conversations with @emilyjablonski about this decision if there's any thoughts she'd like to add here.

Reply:
Yeah we've had a few chats w design about filtering on the same page vs having separate pages, and the separate pages was requested from design for the reasons Colin mentioned above

Reply:
Agreed though that the filtering would be a better UX